src/v/cluster/partition_balancer_planner.h

```diff
@@ -80,9 +74,18 @@ class partition_balancer_planner {
 
         absl::flat_hash_map<model::ntp, size_t> ntp_sizes;
 
-        // Partitions that are planned to move in current planner request
-        absl::flat_hash_set<model::ntp> moving_partitions;
+        absl::btree_map<model::ntp, allocated_partition> reassignments;
```

Comment:
are we interested in order here, i think we agreed that the `node_hash_map` is better for general unordered use cases